src/server/game/Entities/Unit/Unit.cpp

```diff
@@ -16917,6 +16917,11 @@ void Unit::_EnterVehicle(Vehicle* vehicle, int8 seatId, AuraApplication const* a
         }
     }
 
+    // For vehicles, check if there is unit type non move
+    // If so, root the vehicle once someone gets in
+    if (HasFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_DISABLE_MOVE))
```

Comment:
This is strange, in the comment you mention the vehicle but 'this' is the passenger here - checking if passenger has UNIT_FLAG_DISABLE_MOVE won't do any good